nikola/nikola.py

```diff
@@ -737,6 +734,33 @@ def __init__(self, **config):
             utils.LOGGER.warn('The DISABLE_INDEXES_PLUGIN_RSS_FEED setting was renamed to DISABLE_MAIN_RSS_FEED.')
             self.config['DISABLE_MAIN_RSS_FEED'] = self.config['DISABLE_INDEXES_PLUGIN_RSS_FEED']
 
+        if self.config.get('POSTS_SECTIONS'):
+            utils.LOGGER.warn("The sections feature has been removed and its functionality has been merged into categories.")
+            utils.LOGGER.warn("For more information on how to migrate, please read: TODO")  # TODO: blog post
+
+            for section_config_suffix, cat_config_suffix in (
+                ('DESCRIPTIONS', 'DESCRIPTIONS'),
+                ('TITLE', 'TITLES'),
+                ('TRANSLATIONS', 'TRANSLATIONS')
+            ):
+                section_config = 'POSTS_SECTION_' + section_config_suffix
+                cat_config = 'CATEGORY_' + cat_config_suffix
+                if section_config in self.config:
+                    self.config[section_config].update(self.config[cat_config])
+                    self.config[cat_config] = self.config[section_config]
+
+            self.config['CATEGORY_DESTPATH_NAMES'] = self.config.get('POSTS_SECTION_NAME', {})
+            # Need to mark this translatable manually.
+            self.config['CATEGORY_DESTPATH_NAMES'] = utils.TranslatableSetting('CATEGORY_DESTPATH_NAMES', self.config['CATEGORY_DESTPATH_NAMES'], self.config['TRANSLATIONS'])
+
+            self.config['CATEGORY_DESTPATH_AS_DEFAULT'] = not self.config.get('POSTS_SECTION_FROM_META')
+            utils.LOGGER.info("Setting CATEGORY_DESTPATH_AS_DEFAULT = " + str(self.config['CATEGORY_DESTPATH_AS_DEFAULT']))
+
+        if self.config.get('CATEGORY_PAGES_FOLLOW_DESTPATH') and (not self.config.get('CATEGORY_ALLOW_HIERARCHIES') or self.config.get('CATEGORY_OUTPUT_FLAT_HIERARCHY')):
+            utils.LOGGER.warn('CATEGORY_PAGES_FOLLOW_DESTPATH requires CATEGORY_ALLOW_HIERARCHIES = True, CATEGORY_OUTPUT_FLAT_HIERARCHY = False.')
```

Comment:
Why not make this an error and exit? Or explicitly state that the user-supplied values of `CATEGORY_ALLOW_HIERARCHIES` and `CATEGORY_OUTPUT_FLAT_HIERARCHY` are ignored?